main.go

```diff
@@ -308,111 +153,20 @@ func installVersion(arg string, custBinPath *string, mirrorURL *string) {
-func checkTFEnvExist() bool {
-	tfversion := os.Getenv("TF_VERSION")
-	if tfversion != "" {
-		return true
-	}
-	return false
-}
-
-/* parses everything in the toml file, return required version and bin path */
-func getParamsTOML(binPath string, dir string) (string, string) {
-	path, err := homedir.Dir()
-
-	if err != nil {
-		fmt.Printf("Unable to get home directory: %v\n", err)
+		fmt.Println("Arg:", arg)
+		param_parsing.UsageMessage()
 		os.Exit(1)
 	}
-
-	if dir == path {
-		path = "home directory"
-	} else {
-		path = "current directory"
-	}
-	fmt.Printf("Reading configuration from %s\n", path+" for "+tomlFilename) //takes the default bin (defaultBin) if user does not specify bin path
-	configfileName := lib.GetFileName(tomlFilename)                          //get the config file
-	viper.SetConfigType("toml")
-	viper.SetConfigName(configfileName)
-	viper.AddConfigPath(dir)
-
-	errs := viper.ReadInConfig() // Find and read the config file
-	if errs != nil {
-		fmt.Printf("Unable to read %s provided\n", tomlFilename) // Handle errors reading the config file
-		fmt.Println(errs)
-		os.Exit(1) // exit immediately if config file provided but it is unable to read it
-	}
-
-	bin := viper.Get("bin")                                                     // read custom binary location
-	if binPath == lib.ConvertExecutableExt(lib.GetDefaultBin()) && bin != nil { // if the bin path is the same as the default binary path and if the custom binary is provided in the toml file (use it)
-		binPath = os.ExpandEnv(bin.(string))
-	}
-	//fmt.Println(binPath) //uncomment this to debug
-	version := viper.Get("version") //attempt to get the version if it's provided in the toml
-	if version == nil {
-		version = ""
-	}
-
-	return version.(string), binPath
-}
-
-func usageMessage() {
-	fmt.Print("\n\n")
-	getopt.PrintUsage(os.Stderr)
-	fmt.Println("Supply the terraform version as an argument, or choose from a menu")
 }
 
 /* installOption : displays & installs tf version */
 /* listAll = true - all versions including beta and rc will be displayed */
 /* listAll = false - only official stable release are displayed */
-func installOption(listAll bool, custBinPath, mirrorURL *string) {
-	tflist, _ := lib.GetTFList(*mirrorURL, listAll) //get list of versions
-	recentVersions, _ := lib.GetRecentVersions()    //get recent versions from RECENT file
-	tflist = append(recentVersions, tflist...)      //append recent versions to the top of the list
-	tflist = lib.RemoveDuplicateVersions(tflist)    //remove duplicate version
+func installOption(listAll bool, customBinaryPath, mirrorURL string) {
+	tflist, _ := lib.GetTFList(mirrorURL, listAll) //get list of versions
+	recentVersions, _ := lib.GetRecentVersions()   //get recent versions from RECENT file
+	tflist = append(recentVersions, tflist...)     //append recent versions to the top of the list
+	tflist = lib.RemoveDuplicateVersions(tflist)   //remove duplicate version
```

Comment:
For the sake of consistency
```suggestion
	tflist, _ := lib.GetTFList(mirrorURL, listAll) // Get list of versions
	recentVersions, _ := lib.GetRecentVersions()   // Get recent versions from RECENT file
	tflist = append(recentVersions, tflist...)     // Append recent versions to the top of the list
	tflist = lib.RemoveDuplicateVersions(tflist)   // Remove duplicate version
```